File: CivitasJava/civitas/SorpresaIrCarcel.java
package civitas;

import java.util.ArrayList;

public class SorpresaIrCarcel extends Sorpresa {
    private Tablero tablero;
    private int numCasillaCarcel;
    
    SorpresaIrCarcel(Tablero tab, int numCasillaCarcel, String tex){
        super(tex);
        tablero = tab;
        this.numCasillaCarcel = numCasillaCarcel;
    }
    
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos) throws Exception{
        if ( jugadorCorrecto(actual, todos) ){
            informe(actual, todos);
            todos.get(actual).encarcelar( tablero.getCarcel() );
        }       
    }
}

File: CivitasRuby/sorpresa_ir_carcel.rb
require_relative 'sorpresa'
require_relative 'tablero'

module Civitas
class SorpresaIrCarcel < Sorpresa
  
  def initialize(tablero, carcel, texto)
    super(texto)
    @tablero = tablero
    @numCasillaCarcel = carcel
  end
  
  def aplicar_a_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      informe(actual, todos)
      todos[actual].encarcelar(@tablero.numCasillaCarcel)
    end
  end
  
  public_class_method :new
end
end
